emoji-ui/src/components/emojiItem.js

```diff
@@ -6,17 +6,18 @@ class EmojiItem extends Component {
   static propTypes = {
     isSelected: PropTypes.bool.isRequired,
     url: PropTypes.string.isRequired,
-    clickHandler: PropTypes.func.isRequired.isRequired,
+    name: PropTypes.string.isRequired,
+    clickHandler: PropTypes.func.isRequired,
   };
 
   render() {
-    const { isSelected, url, clickHandler } = this.props;
+    const { isSelected, url, name, clickHandler } = this.props;
     const styles = { backgroundImage: `url(${url})` };
-    const containerClasses = `Emoji-container${isSelected ? ' Emoji-container-selected' : ''}`;
+    const containerClasses = `emoji-container${isSelected ? ' emoji-container--selected' : ''}`;
 
     return (
       <div className={containerClasses}>
-        <button style={styles} className='Emoji-item' onClick={clickHandler}></button>
+        <button style={styles} className='emoji-item' onClick={() => clickHandler(name)}></button>
```

Comment:
for future reference: this way of writing a handler wrapper will create a new anonymous function every time render() is called (so any time props/state) change e.g. when an item is selected/deselected. if you instead bind a local function in the constructor, you can avoid this. the constructor-binding way is wordier, though, and there is unlikely to be enough gain to argue for it in a case like this. at some point it comes down to personal preference and style. the way you have it now is simpler.

Reply:
also ^ this is just a comment, not a request for change. i'm 100% good with it like this because simple is nice. i just wanted to explain why my personal style is to never do this on the merchant app repo.

Reply:
good to know! thanks!